extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/RemoteSyncHandler.java

```diff
@@ -0,0 +1,235 @@
+
+        synchronized (RemoteSyncHandler.class) {
+            checkForChanges = true;
+            //if there is a current dev request this will unblock it
+            RemoteSyncHandler.class.notifyAll();
+            try {
+                RemoteSyncHandler.class.wait(30000);
+            } catch (InterruptedException e) {
+                log.error("interrupted", e);
+            }
+        }
+    }
+
+    @Override
+    public void handle(HttpServerRequest event) {
+        long time = System.currentTimeMillis();
+        if (time > currentSessionTimeout) {
+            currentSession = null;
+        }
+        final String type = event.headers().get(HttpHeaderNames.CONTENT_TYPE);
+        if (APPLICATION_QUARKUS.equals(type)) {
+            String rp = event.headers().get(QUARKUS_PASSWORD);
+            if (!password.equals(rp)) {
+                log.error("Incorrect password");
+                event.response().setStatusCode(401).end();
+                return;
+            }
+            currentSessionTimeout = time + 60000;
+            ExecutorRecorder.getCurrent().execute(new Runnable() {
+                @Override
+                public void run() {
+                    handleRequest(event);
+                }
+            });
+            return;
+        }
+        next.handle(event);
+    }
+
+    private void handleRequest(HttpServerRequest event) {
+        if (event.method().equals(HttpMethod.PUT)) {
+            handlePut(event);
+        } else if (event.method().equals(HttpMethod.DELETE)) {
+            handleDelete(event);
+        } else if (event.method().equals(HttpMethod.POST)) {
+            if (event.path().equals(DEV)) {
+                handleDev(event);
+            } else if (event.path().equals(CONNECT)) {
+                handleConnect(event);
+            } else {
+                event.response().setStatusCode(404).end();
+            }
+        } else {
+            event.response().setStatusCode(404).end();
+        }
+
+    }
+
+    private void handleDev(HttpServerRequest event) {
+        if (checkSession(event))
+            return;
+        event.bodyHandler(new Handler<Buffer>() {
+            @Override
+            public void handle(Buffer b) {
+                ExecutorRecorder.getCurrent().execute(new Runnable() {
+                    @Override
+                    public void run() {
+                        try {
+                            Throwable problem = (Throwable) new ObjectInputStream(new ByteArrayInputStream(b.getBytes()))
+                                    .readObject();
+                            if (problem != null) {
+                                remoteProblem = problem;
+                                hotReplacementContext.setRemoteProblem(problem);
+                            } else if (remoteProblem != null) {
+                                remoteProblem = null;
+                                hotReplacementContext.setRemoteProblem(problem);
```

Comment:
Shouldn't it be `remoteProblem` instead of the `problem` which is null here?
Also at line 129 you are assigning `remoteProblem` to `problem` but in this block you are setting `remoteProblem` to null. Is that how it is supposed to be? Looks confusing.

Reply:
It should be an explicit null.